src/debug/ee/debugger.cpp

```diff
@@ -898,6 +899,7 @@ AbortTransport()
     {
         g_pDbgTransport->AbortConnection();
     }
+    DiagnosticServer::Shutdown();
```

Comment:
I'm skeptical this is the best place to insert this call. Conceptually the diagnostics server has no dependency on the debugger, they are two separate components that both happen to use a pipe. I would look up the callstack for the error handling code that is outside the debugger and hook the diagnostics server shutdown code there.

Reply:
Thanks for the feedback Noah! I just updated the PR with your suggestion :)